### ISSUE TITLE
generateSuppressAllWarnings is ignored if base type has SuppressAllWarnings

### ISSUE THREAD

**Issue by jfancher** [immutables/immutables#393]
Admittedly this is a very obscure usage. I found it when trying to find a local solution to #392.

Here's an example:
````java
@Value.Immutable
@SuppressWarnings({"all", "varargs"})
@Value.Style(generateSuppressAllWarnings = false)
interface Bug<T> {}
````

The generated type has `@SuppressWarnings("all")` on it despite the style setting. Unfortunately, it does _not_ have `@SuppressWarnings({"all", "varargs"})`. I would expect either nothing (respecting generateSuppressAllWarnings) or both (copying the annotation).

Possibly related: `passAnnotations = {SuppressWarnings.class}` in the Style does not seem to work in any case.

**Comment by elucash** [comment]
`passAnnotations` ignores it because it's `java.lang.*` annotation. Copying suppress warnings values to the generated class is probably a good idea, thanks!

**Comment by elucash** [comment]
Released in 2.2.11 Thank you!
